crates/resolver/src/recursor/mod.rs

```diff
@@ -90,6 +104,61 @@ impl Recursor<TokioRuntimeProvider> {
 }
 
 impl<P: ConnectionProvider> Recursor<P> {
+    /// Build a new [`Recursor`] from the specified configuration
+    pub fn from_config(
```

Comment:
I think we should probably flatten just to drop the builder and go straight from config to a `Recursor`? The builder adds a lot of boilerplate and is mostly trivial for this type. Didn't want to keep that in this PR, though, which is already large enough.